code/parse.go

```diff
@@ -223,11 +225,21 @@ func parseCloudTrailLogs(data events.CloudwatchLogsData) []ingest.Log {
 			}
 		} else if eventSource == "ecs.amazonaws.com" {
 			ecsStreamRegex, _ := regexp.Compile(`("cluster":"|"cluster": "|:cluster/)([^/][^,][^"]*)`)
+
 			ecsStreamArray := ecsStreamRegex.FindStringSubmatch(event.Message)
+
 			if len(ecsStreamArray) > 2 {
 				resoureIDMap["system.aws.arn"] = fmt.Sprintf("arn:aws:ecs:%s:%s:cluster/%s", awsRegion, data.Owner, ecsStreamArray[2])
 				accountLevelLog = false
 			}
+		} else if eventSource == "s3.amazonaws.com" {
+			s3RegexArray := s3Regex.FindStringSubmatch(event.Message)
+			s3Arn := s3Regex.SubexpIndex("arn")
```

Comment:
Do we really need regex to extract arn, can't we do it without ?

Reply:
@choudharypooja Converting to json would even be a costlier operation .. in comparison the regex should be better I guess.

Reply:
Some benchmarks would be helpful, regex can also be computationally expensive, since it has to iterate through the white body.
I guess it is fine for now.

Reply:
Yes we can... I guess at all places regex is used here... We can may be look for a common approach.

Reply:
Since we have move the regex compilation to the global scope to have it computed once.
Feels weird that others regex are in function's scope, Would be great to do that for all regex.